tasks/update-infra-deployments.yaml

```diff
@@ -23,7 +29,7 @@ spec:
           # 'appstudio-staging-ci' - private key for appstudio-staging-ci Github app
```

Comment:
There is an issue with sharedSecret, you will need to create custom one, since it requires deploy key must be unique for each project. Another (better) option is to use [github api contents](https://docs.github.com/en/rest/repos/contents#create-or-update-file-contents) , so then we would not need deploy key.

Reply:
having it in progress

Reply:
Oh interesting. I didn't know about that approach. Will that create a PR to be reviewed? Or just push a commit directly?

Reply:
So basically there are two secrets one is deploy key which is project related, used to create branch and push the commits (using ssh key) and second is secret for github app to be able to create Pull Request.
Now I am trying to rewrite code to use github api contents instead of standard git commands. Prototype done, trying to inject to the task.

Reply:
PR is rebased on the change that implements the new option.